core/src/main/java/org/apache/calcite/rex/RexBuilder.java

```diff
@@ -1345,6 +1384,10 @@ private static Comparable zeroValue(RelDataType type) {
       return new TimeString(0, 0, 0);
     case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
       return new TimestampString(0, 0, 0, 0, 0, 0);
+    case TIME_WITH_TIME_ZONE:
+      return new TimeWithTimeZoneString(0, 0, 0, "UTC");
+    case TIMESTAMP_WITH_TIME_ZONE:
+      return new TimestampWithTimeZoneString(0, 0, 0, 0, 0, 0, "UTC");
```

Comment:
Shouldn't  0 is `January 1, 1970 UTC`?

Reply:
After double checked the ZERO of TIMESTAMP type, I think your are right. ZERO should be `January 1, 1970 UTC`.